CRM/MembershipExtras/Service/MembershipTypeInstalmentCalculator.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+use CRM_MembershipExtras_Service_MoneyUtilities as MoneyUtilities;
+use CRM_MembershipExtras_Service_MembershipTypeTaxAmountCalculator as MembershipTypeTaxAmountCalculator;
+use CRM_MembershipExtras_Exception_InvalidMembershipTypeInstalmentCalculator as InvalidMembershipTypeInstalmentCalculator;
+use CRM_MembershipExtras_Service_MembershipTypeDatesCalculator as MembershipTypeDatesCalculator;
+use CRM_MembershipExtras_DTO_ScheduleInstalmentAmount as ScheduleInstalmentAmount;
+
+/**
+ * Class CRM_MembershipExtras_Service_MembershipTypeInstalmentCalculator
+ */
+class CRM_MembershipExtras_Service_MembershipTypeInstalmentCalculator {
```

Comment:
I would rename this class maybe to : 

CRM_MembershipExtras_Service_MembershipInstalmentsSchedule

and rename  **generateInstalments** to **generate**

then I would reorder the methods in the order they are called in starting with the public method.